draft-ietf-tls-esni.md

```diff
@@ -459,27 +459,27 @@ it.
 
 To prevent a network attacker from modifying the reconstructed ClientHelloInner
 (see {{flow-clienthello-malleability}}), ECH authenticates ClientHelloOuter by
-computing ClientHelloOuterAAD as described below and passing it in as the
-associated data for HPKE sealing and opening operations. ClientHelloOuterAAD has
-the following structure:
-
-~~~
-   struct {
-      HpkeSymmetricCipherSuite cipher_suite;
-      uint8 config_id;
-      opaque enc<1..2^16-1>;
-      ClientHello outer_hello;
-   } ClientHelloOuterAAD;
-~~~
-
-The first three parameters are equal to, respectively, the
-`ClientECH.cipher_suite`, `ClientECH.config_id`, and `ClientECH.enc` fields of
-the payload of the "encrypted_client_hello" extension. The last parameter,
-`outer_hello`, is computed by serializing ClientHelloOuter with the
-"encrypted_client_hello" extension set to the empty string, i.e., the
-`extension_data` list has zero length. This serialization uses the
-ClientHello structure from {{Section 4.1.2 of RFC8446}}, which does not include
-the four-byte header included in the Handshake structure.
+passing ClientHelloOuterAAD as the associated data for HPKE sealing and opening
+operations. The ClientHelloOuterAAD is a serialized ClientHello structure,
+defined in {{Section 4.1.2 of RFC8446}}, which matches the ClientHelloOuter
+except the `payload` field of the "encrypted_client_hello" is replaced with a
+byte string of the same length but whose contents are zeros. This value does
+not include the four-byte header from the Handshake structure.
+
+The client constructs ClientHelloOuterAAD first, followed by ClientHelloOuter.
+First, the client predicts the length of the `payload` based on the HPKE AEAD
```

Comment:
I'm kind of thrown off by the use of the word "predict." It suggests to me that there's a chance that the prediction might be wrong. Should it not be "compute" instead? Or is there in fact a chance it might be wrong?